pkg/cluster/check/nodes.go

```diff
@@ -0,0 +1,236 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package check
+
+import (
+	"context"
+	"errors"
+	fmt "fmt"
+
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/safe"
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/dustin/go-humanize"
+	"github.com/hashicorp/go-multierror"
+	"github.com/siderolabs/gen/maps"
+	"github.com/siderolabs/gen/slices"
+	"google.golang.org/grpc/codes"
+
+	"github.com/siderolabs/talos/pkg/machinery/client"
+	"github.com/siderolabs/talos/pkg/machinery/config/types/v1alpha1/machine"
+	"github.com/siderolabs/talos/pkg/machinery/constants"
+	"github.com/siderolabs/talos/pkg/machinery/resources/runtime"
+	"github.com/siderolabs/talos/pkg/machinery/resources/v1alpha1"
+	"github.com/siderolabs/talos/pkg/minimal"
+)
+
+// AllNodesMemorySizes checks that all nodes have enough memory.
+func AllNodesMemorySizes(ctx context.Context, cluster ClusterInfo) error {
+	cl, err := cluster.Client()
+	if err != nil {
+		return fmt.Errorf("error getting client: %w", err)
+	}
+
+	nodes := getNodesTypes(cluster, machine.TypeInit, machine.TypeControlPlane, machine.TypeWorker)
+
+	resp, err := cl.Memory(client.WithNodes(ctx, maps.Keys(nodes)...))
+	if err != nil {
+		return fmt.Errorf("error getting nodes memory: %w", err)
+	}
+
+	var resultErr error
+
+	for _, msg := range resp.Messages {
+		if msg.Metadata == nil {
+			return fmt.Errorf("no metadata in the response")
+		}
+
+		hostname := msg.Metadata.Hostname
+
+		typ, ok := nodes[hostname]
+		if !ok {
+			return fmt.Errorf("unexpected node %q in response", hostname)
+		}
+
+		if msg.Metadata.Error != "" {
```

Comment:
this is not needed, `Memory` function of the client does error unwrapping removing any `Metadata.Error` responses and pushing them to `err error` response

Reply:
Fixed!